containers/init/ready/ready.go

```diff
@@ -178,10 +203,18 @@ func WaitForReadyPods(ctx context.Context, ltg LoadTestGetter, pl PodLister, tes
 		}
 		ownedPods := status.PodsForLoadTest(loadtest, podList.Items)
 		for _, pod := range ownedPods {
-			if !isPodReady(pod) {
+			if pod.Labels[config.RoleLabel] == config.DriverRole {
+				if !driverMatched && pod.Status.PodIP != "" {
+					nodesInfo.Driver = NodeInfo{
+						Name:     pod.Name,
+						PodIP:    pod.Status.PodIP,
+						NodeName: pod.Spec.NodeName,
+					}
+					driverMatched = true
+				}
```

Comment:
Why this portion is before checking if the pod is ready? Part of the isPodReady part is checking if the IP exist.

Reply:
Yeah, good question. since ready.go is running in driver's ready init container, itself would not be marked as ready. So in order to fetch driver's ip and node name, we have to grab it before ready check.